lib/src/resolver.dart

```diff
@@ -158,9 +158,9 @@ class Loader {
 
   /// Loads an imported resource and returns a [Future] with a [List] of lines.
   /// Returns `null` if the resource could not be loaded.
-  Future<List<String>> load(String path) async {
+  List<String> load(String path) {
```

Comment:
The existing API which returns a `Future` of the lines in the source file is an intentional design decision to allow for an HTTP-based loader that loads files from the VM service port. Dart doesn't provide synchronous socket APIs, so an async API in required in such cases.

Reply:
A potential improvement over the current API would be to switch to a `Stream<String>`, then read the file in as a byte-stream, and run it through a `UTF8Decoder` and a `LineSplitter`.